### PR TITLE
add RNDADDENTROPY for ioctl interface on linux glibc.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
The value seems different on powerpc:
```
 bad RNDADDENTROPY value at byte 0: rust: 64 (0x40) != c 128 (0x80)
```

**Comment by workingjubilee** [comment]
Right, it's this: https://github.com/torvalds/linux/blob/6bb20c152b6bf7dd8ffb248f33c2593fd9aeb318/include/uapi/linux/random.h#L30

But that uses an ioctl macro to expand to the "magic" number here.

**Comment by JohnTitor** [comment]
The failure is:
```
error[E0432]: unresolved import `libc::RNDADDENTROPY`
    --> /checkout/target/powerpc-unknown-linux-gnu/debug/build/libc-test-86e0293debea6117/out/semver.rs:4909:5
     |
4909 |     RNDADDENTROPY,
     |     ^^^^^^^^^^^^^ no `RNDADDENTROPY` in the root
```

@rustbot author

File: libc-test/build.rs [hunk 1]
```diff
@@ -3905,7 +3905,8 @@ fn test_linux(target: &str) {
         // the `ifr_ifru` field is an anonymous union
         (struct_ == "ifreq" && field == "ifr_ifru") ||
         // glibc uses a single array `uregs` instead of individual fields.
-        (struct_ == "user_regs" && arm)
+        (struct_ == "user_regs" && arm) ||
+        (struct_ == "rand_pool_info" && field == "buf")
```
Comment (JohnTitor) [inline libc-test/build.rs:3909]: The comment should be added like the above.

**Review by JohnTitor — APPROVED** [review]

**Comment by workingjubilee** [comment]
```
    Compiling libc-test v0.2.142 (/checkout/libc-test)
error[E0432]: unresolved import `libc::RNDADDENTROPY`
    --> /checkout/target/arm-unknown-linux-gnueabihf/debug/build/libc-test-8a7c80780f2f9057/out/semver.rs:4764:5
     |
4764 |     RNDADDENTROPY,
     |     ^^^^^^^^^^^^^ no `RNDADDENTROPY` in the root
```

???

@rustbot author

**Comment by JohnTitor** [comment]
Closing as inactive, feel free to re-submit a PR if you're still interested in this change. Thanks anyway!

**Pull request closed without merging**